2024/03-extensions/dist-and-networks-slides.qmd

```diff
@@ -0,0 +1,418 @@
+
+```{r, echo = T}
+penguins_brm |>
+tidybayes::gather_draws(
+c(b_Intercept,
+b_speciesChinstrap,
+b_speciesGentoo)) %>%
+ggplot(aes(y = .variable, x = .value)) +  geom_dotsinterval(side = "bottom", dotsize = .05) + 
+stat_slabinterval() 
+```
+
+## Takeaways
+
+-   `ggdist` contains `geom` and `stat` functions for plotting distributions.
+-   Slab functions show variations on density plots.
+-   Ribbons show variation around a line.
+-   `ggdist` plays nicely with Bayesian and frequentist modeling frameworks.
+
+## For more
+
+- [`ggdist` documentation here](https://mjskay.github.io/ggdist/index.html)
+- [`tidybayes` documentation here](http://mjskay.github.io/tidybayes/)
+
+# Relational data with `ggraph`
+
+## `ggraph` {.smaller}
+
+-   For plotting networks, graphs, and trees
+-   Adds **layouts** and geoms for nodes and edges
+-   Integrates with [`tidygraph`](https://ggraph.data-imaginist.com/articles/tidygraph.html) to wrap around numerous other graph object types (`igraph`, `dedrogram`, `hclust`, `graph`, `phylo`, ...)
+
+## Example: Mouse data
+
+Begin by loading network data, in this case from the `graphml` format.
+
+Data from [So et al. 2015](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0134509), in the [Animal Social Networks Repository](https://github.com/bansallab/asnr/tree/master).
+
+```{r, echo = TRUE}
+
+library(igraph)
+library(ggraph)
+library(tidygraph)
+
+mouse_sniffing <- read_graph(here::here("2024", "03-extensions", "mouse_so_grooming_network.graphml"),
+format = "graphml")
+
+```
+
+## Example: `as_tbl_graph`
+
+`tidygraph::as_tbl_graph` standardizes different graph storage formats.
+
+```{r, echo = T}
+mouse_graph <- mouse_sniffing |>
+as_tbl_graph() 
+```
+
+## Plotting mouse sniffing data
+
+```{r, echo = T}
+ggraph(mouse_graph, layout = 'kk') + 
+    geom_edge_fan() + 
+    geom_node_point() +
+    theme_minimal() +
+    ggtitle("Mouse sniffs")
+```
+
+## Alternate layouts
+
+```{r, echo = T}
+ggraph(mouse_graph, layout = 'eigen') + 
+    geom_edge_fan() + 
+    geom_node_point() +
+    theme_minimal() +
+    ggtitle("Mouse sniffs")
+```
+
+## Alternate layouts
+
+```{r, echo = T}
+ggraph(mouse_graph, layout = 'linear', circular = TRUE) + 
+    geom_edge_fan() + 
+    geom_node_point() +
+    theme_minimal() +
+    ggtitle("Mouse sniffs")
+```
+
+## Example: phylogenies
+
+-   `ggraph` can plot phylogenetic trees.
+-   We'll get a phylogeny using the R Open Tree of Life (`rotl`) and plot it.
+
+## Getting a phylogeny
+
+![](krat.jpg){fig-align="center"}
+
+## Getting a phylogeny
+
+```{r, echo = TRUE, warning = F}
+library(rotl)
+rodent_id <- tnrs_match_names("Heteromyidae")
+rodent_tree <- tol_subtree(ott_id = ott_id(rodent_id))
+rodent_tree$node.label <- c(1:64)
+class(rodent_tree)
+```
+
+## Convert the phylogeny to a graph
+
+```{r, echo = T}
+
+rodent_graph <- as_tbl_graph(rodent_tree)
+
+```
+
+## Dendrogram
+
+```{r, echo = T}
+
+ggraph(rodent_graph, "dendrogram") +
+geom_edge_link()
```

Comment:
Ugh, there is not. In fact, the branch tips labeling breaks the conversion from `phlyo` to `graph`, and I manually stripped them out. This might be a reason not to include this example?

Reply:
Not necessarily, just trying to anticipate questions.  I feel like the way it _should_ work is just with `geom_text()` or maybe like a `geom_tip_text()`, but assumed it might not be so simple.

Reply:
I think it would be simple if it were simple to keep the labels during the conversion process. I couldn't figure it out yesterday - I'd get a "Duplicated node labels" error when there were no duplicates, and no hits on Google.